content/openusercss-integration.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+(() => {
+  const manifest = chrome.runtime.getManifest();
+
+  // Tell the page that we exist and that it should send the handshake
+  window.postMessage({
+    'type': 'ouc-begin-handshake'
+  }, '*');
+
+  // Wait for the handshake
+  window.addEventListener('message', event => {
+    if (event.data && event.data.type === 'ouc-handshake-question') {
+      // This is a representation of features that Stylus is capable of
+      const implementedFeatures = [
+        'install-usercss',
+        'configure-after-install',
+        'builtin-editor',
+        'create-usercss',
+        'edit-usercss',
+        'import-moz-export',
+        'export-moz-export',
+        'update-manual',
+        'update-auto',
+        'export-json-backups',
+        'import-json-backups',
+        'manage-local'
+      ];
+      const reportedFeatures = [];
+
+      // The handshake question includes a list of required and optional features
+      // we match them with features we have implemented, and build a union array.
+      event.data.featuresList.required.forEach(feature => {
+        if (implementedFeatures.includes(feature)) {
+          reportedFeatures.push(feature);
+        }
+      });
+
+      event.data.featuresList.optional.forEach(feature => {
+        if (implementedFeatures.includes(feature)) {
+          reportedFeatures.push(feature);
+        }
+      });
+
+      // We send the handshake response, which includes the key we got, plus some
+      // additional metadata
+      window.postMessage({
+        'type':      'ouc-handshake-response',
+        'key':       event.data.key,
+        'extension': {
+          'name':         manifest.name,
+          'version':      manifest.version,
```

Comment:
There shouldn't be any need for the extension's manifest, it's irrelevant to the functionality.

Reply:
I don't think it's irrelevant, because the site shows these values: 
In the install button: ![screenshot](https://i.imgur.com/0IEJCNW.png)
On the help page: ![screenshot](https://i.imgur.com/Pi8xdlP.png)

Reply:
The users don't like being tracked. There's no need to check for the version. As you know, it's always preferable to test for specific features like you already do.

Reply:
I don't see how showing the version number on the help page could be seen as tracking, but yes, there's not a critical need for it.
I made it optional for extensions in the client javscript and removed the version number from the response object.

Reply:
Users are that weird. Not sure why you're still using manifest though. The extension name shouldn't serve any useful purpose - the capabilities is the only meaningful object there.

Reply:
The site doesn't assume the extension name, so it has to get it from somewhere.
The install style button is displaying "Install theme with {{extension.name}}".

Reply:
Ah, makes sense then.